src/startProductRelease.ts

```diff
@@ -0,0 +1,291 @@
+    codeBaseName: 'product-api',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-approvals-api',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-approvals-client',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-cognito-hosted-login-css',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-console',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-form-store-client',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-forms-lambda-at-edge-authorisation',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-forms-renderer',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-infrastructure',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-pdf',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-s3-submission-events',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'product-volunteers-client',
+    type: 'NODE_JS',
+  },
+  {
+    codeBaseName: 'sdk-core-js',
+    type: 'NPM',
+  },
+  {
+    codeBaseName: 'sdk-dotnet',
+    type: 'NUGET',
+  },
+  {
+    codeBaseName: 'sdk-node-js',
+    type: 'NPM',
+  },
+]
+
+export default async function startProductRelease({
+  releaseName,
+}: {
+  releaseName: string
+}) {
+  console.log('Beginning Product release process for:', releaseName)
+
+  const deploymentRequiredUrls: string[] = []
+  for (const { codeBaseName, type } of gitCloneCodeBases) {
+    const codeBaseWorkingDirectory = await wrapWithLoading(
+      {
+        startText: `Creating temporary directory to clone "${codeBaseName}"`,
+        failText: `Failed to create temporary directory to clone "${codeBaseName}"`,
+      },
+      async (spinner) => {
+        const directory = await mkdtemp(join(tmpdir(), codeBaseName))
+        spinner.succeed(
+          `Created temporary directory to clone "${codeBaseName}"`,
+        )
+        return directory
+      },
+    )
+
+    try {
+      const cloneUrl = `git@github.com:oneblink/${codeBaseName}.git`
+      await executeCommand(
+        'git',
+        ['clone', cloneUrl, codeBaseWorkingDirectory],
+        '.',
+      )
+
+      // Check if code base needs releasing
+      const { parsedChangelog } = await parseChangelogWithLoading(
+        codeBaseWorkingDirectory,
+      )
+      const unreleasedVersion = parsedChangelog.versions.find((version) =>
+        version.title.toLowerCase().includes('unreleased'),
+      )
+      if (!unreleasedVersion) {
+        await continuePromptWithWarning(`"${codeBaseName}" CHANGELOG.md does not contain an "Unreleased" section
+
+You need to checkout "${cloneUrl}" to fix this before trying again.`)
+        continue
+      }
+
+      const unreleasedChangelogEntries = unreleasedVersion.body.trim()
+      if (!unreleasedChangelogEntries) {
+        ora(
+          `Skipping "${codeBaseName}" as CHANGELOG.md does not contain any entries in the "Unreleased" section.`,
+        )
+          .start()
+          .info()
+        continue
+      }
```

Comment:
From my understanding, the dependency checker that adds to the changelog runs in `startReleaseProcess()`/at a later point, would this mean that any repo that only had dependency updates (e.g Forms-CDN, Approvals-Client, etc.) would be skipped?

Reply:
perhaps another way of evaluating this is checking if the last commit on the default branch is after the commit from the last release? It seems a little fragile relying on changelog entries.

Reply:
Yep, you are right @jdawg093.

I was trying to avoid checking commits as we have CI jobs that do commits after release commits for documentation. We may may have others in the future. This is why I deemed it fragile relying on commits and seemed more reliable to skip releases if there are no changelog entries.

I will keep thinkering.